src/citrine/resources/material_spec.py

```diff
@@ -105,12 +112,43 @@ def filter_by_template(self,
         :param per_page: The number of results to return per page.
         :return: A search result of material specs
         """
-        path_prefix = MaterialTemplateCollection(self.project_id,
-                                                 self.dataset_id,
-                                                 self.session)._get_path(ignore_dataset=True)
-        path = path_prefix + "/" + template_scope + "/" + template_id + "/material-specs"
-        raw_objects = self.session.cursor_paged_resource(self.session.get_resource,
-                                                         path,
-                                                         per_page=per_page,
-                                                         version="v1")
-        return (self.build(raw) for raw in raw_objects)
+        if per_page is not None:
+            logger.warning('The per_page parameter will be ignored. Please remove it.')
+        return self.list_by_template(uid=template_id, scope=template_scope)
+
+    def list_by_template(self, uid: Union[UUID, str], scope: str = 'id') -> Iterator[MaterialSpec]:
```

Comment:
These methods are pretty copypasta-y. Unfortunately, our data model isn't quite consistent enough to be able to add a valid `list_by_template` to `ObjectSpecCollection`.

Reply:
We could have yet another layer of abstract indirection after ObjectSpec/ObjectSpecCollection IngredientSpec/IngredientSpecCollection doesn't inherit from

Reply:
I would be inclined to do that if this method had any complexity to it, but as it is, it's a simple one-liner. An additional class which doesn't reflect a group of object types in our GEMD docs doesn't seem worth it to me to reduce 2 one-line methods to 1 one-line method.